pkg/deploy-manager/subscription.go

```diff
@@ -360,3 +360,136 @@ func (t *DeployManager) waitForOCSOperator() error {
 
 	return nil
 }
+
+// UninstallOCS uninstalls ocs operator and storage clusters
+func (t *DeployManager) UninstallOCS(ocsRegistryImage string, localStorageRegistryImage string) error {
+	// Remove finalizers from all cephclusters to not block the cleanup
+	err := t.removeCephClusterFinalizers()
```

Comment:
At least with the newer Rook versions (v1.1.4 at the time I write this comment), we no longer need to remove the finalizers on CephCluster.

So we can simplify and rearrange the steps here to,

1. Delete Noobaa (and wait for noobaa pv to be released)
2. Delete StorageCluster. This deletes CephCluster and most ceph pods.
3. Delete the remaining ceph and csi driver pods. Delete all daemonsets (for csi nodeplugin pods) and deploy with label '!olm.owner.kind'.
4. Delete OSD pvcs, lable 'ceph.rook.io/DeviceSet'.
5. Delete subscription, csv. Stops the operator pods.
6. Delete namespace.

Reply:
Made the changes. PTAL @kshlm